docs: composition validation (#399)
crossplane/docs: content/master/concepts/composition.md

`None`. Considers the composed resource to be ready as soon as it exists.

### Composition validation

Crossplane uses a `Validating Webhook` to inform users of any potential
errors in a `Composition`. By default webhooks only perform
`logical checks`. `logical checks` enforce requirements that
aren't explicitly defined in the schema but Crossplane assumes to hold at runtime.

#### Experimental validation with schemas

Enable experimental schema-aware validation in Crossplane
through the `--enable-composition-webhook-schema-validation` feature flag. This
enables Composition validation against available schemas in the cluster.
For example, ensuring that `fieldPaths` are valid and source and destination
types match taking into account provided transforms too.

The `crossplane.io/composition-validation-mode` annotation on the Composition
allows setting one of two modes for schema validation:

> -  `loose` (default): Same as `strict` mode, except that in case of missing
>     required schemas, schema validation will be skipped emitting only a
>     warning.

Reviewer: ```suggestion
-  `loose` (default): Validates Compositions against required schemas. If a 
    required schema is missing, schema validation stops, emits a warning and 
    falls back to `logical checks` only.
```

`None`. Considers the composed resource to be ready as soon as it exists.

### Composition Validation

Crossplane uses a `Validating Webhook` to inform users of any potential
misconfigurations in a `Composition` as early as possible. The default behavior
of the webhook is to perform `logical checks` only, enforcing requirements that
aren't explicitly defined in the schema but are assumed to hold at runtime.

#### Experimental Validation With Schemas

Crossplane provides experimental schema-aware validation that can be enabled
through the `--enable-composition-webhook-schema-validation` feature flag. This
will enable Composition validation against available schemas in the cluster,
ensuring, for example, that `fieldPaths` are valid and source and destination
types match taking into account provided transforms too.

The `crossplane.io/composition-validation-mode` annotation on the Composition
allows setting one of two modes for schema validation:

> -  `loose` (default): Same as `strict` mode, except that in case of missing

Reviewer: ```suggestion
-  `loose` (default): Same as `strict` mode, except skips missing required
```
passive voice fix

Author: @plumbis, validation is totally skipped at the moment, not just the "missing required schemas", the new wording makes me the latter.

>     required schemas, schema validation will be skipped emitting only a

Reviewer: ```suggestion
    schemas and emits a
```

    warning.

> -  `strict`: Compositions are validated against required schemas, and rejected

Reviewer: ```suggestion
-  `strict`: Validates Compositions against required schemas, and rejects them
```

>     if any error is found. If any of the required schemas is missing, the

Reviewer: ```suggestion
    when finding errors. Rejects any Compositions missing required schemas.
```

>     Composition will be directly rejected.

Reviewer: ```suggestion
```

See the [Composition Validating Webhook design document][validation-design-doc]
for more information about future development around schema-aware validation.

> #### Disabling Webhooks

Reviewer: ```suggestion
#### Disabling webhooks
```
sentence case titles

> Webhooks are enabled by default, but can be disabled by setting

Reviewer: ```suggestion
Crossplane enables webhookos by default. Turn off webhooks by
```
Remove passive voice and `disabled`

Author: Typo in the suggestion:
```suggestion
Crossplane enables webhooks by default. Turn off webhooks by
```

1. Use a `FromCompositeFieldPath` patch to patch from the 'intermediary' field
   you patched to in step 1 to a field on the destination composed resource.

### Composition Validation

Crossplane uses a `Validating Webhook` to inform users of any potential
misconfigurations in a `Composition` as early as possible. The default behavior
of the webhook is to perform `logical checks` only, enforcing requirements that

> are not explicitly defined in the schema but are assumed to be hold at runtime.

Reviewer 2: ```suggestion
are not explicitly defined in the schema but are assumed to be held at runtime.
```

#### Experimental Validation With Schemas

Crossplane provides experimental schema-aware validation that can be enabled
through the `--enable-composition-webhook-schema-validation` feature flag. This

> will enable Composition validation against available schemas in the Cluster,

Reviewer 2: ```suggestion
will enable Composition validation against available schemas in the cluster,
```

> ensuring, for example, that fieldPaths are valid and source and destination

Reviewer 2: ```suggestion
ensuring, for example, that `fieldPaths` are valid and source and destination
```

types match taking into account provided transforms too.

The `crossplane.io/composition-validation-mode` annotation on the Composition

> allows to set one of two modes for schema validation:

Reviewer 2: ```suggestion
allows setting one of two modes for schema validation.
```

> There are two modes of validation:

Reviewer 2: ```suggestion
The two modes of validation are:
```

Author: slightly reworded, but integrated

-  `strict`: Compositions are validated against required schemas, and rejected
    if any error is found. If any of the required schemas is missing, the
    Composition will be directly rejected.
-  `loose` (default): Same as `strict` mode, except that in case of missing
    required schemas, schema validation will be skipped emitting only a

>     warning.

Reviewer 2: ```suggestion
-  `loose` (default): Same as `strict` mode, except that in case of missing
    required schemas, schema validation will be skipped emitting only a
    warning.
-  `strict`: Compositions are validated against required schemas and rejected
    if errors are found. If any of the required schemas is missing, the
    Composition will be directly rejected.
```

> See the dedicated [design document][validation-design-doc] for more information

Reviewer 2: ```suggestion
See the [Composition Validating Webhook design document][validation-design-doc] for more information
```
